gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/ValidateSchemaAnnotationTask.java

```diff
@@ -173,34 +173,36 @@ private String toNativePath(final String path)
    * This method will check the java class annotation on the class instantiated by a className;
    * It will search if that class has an annotation matching {@link SCHEMA_HANDLER_JAVA_ANNOTATION},
    * if found, this class name will be added to "foundClasses" list.
+   * if exceptions or error detected during instantiation of the class, this method will return without doing anything.
    *
    * @param name the name of class to be search annotation from.
    * @param foundClasses a list of class names of the classes that contains {@link SCHEMA_HANDLER_JAVA_ANNOTATION}
-   * @return whether this
-   * @throws IOException
    */
-  private void checkHandlerAnnotation(String name, List<String> foundClasses) throws IOException
+  private void checkHandlerAnnotation(String name, List<String> foundClasses)
   {
     if (name.endsWith(CLASS_SUFFIX))
     {
       int end = name.lastIndexOf(CLASS_SUFFIX);
       String clazzPath = name.substring(0, end);
       String clazzName = pathToName(clazzPath);
 
+      Class<?> clazz = null;
       try
       {
-        Class<?> clazz = classForName(clazzName);
-        for (Annotation a : clazz.getAnnotations())
+        clazz = classForName(clazzName);
+      } catch (Exception | Error e)
+      {
+        getProject().getLogger().info("Unexpected exceptions or errors [{}] found during instantiating the class, will skip this class: [{}]", e.getClass(), clazzName);
+        getProject().getLogger().debug("Unexpected exceptions or errors found during instantiating the class [{}], detailed error: ", clazzName, e);
```

Comment:
Why do we need both info and debug logs? One is not enough?

Reply:
usually people don't open "--debug" flag, so I don't want them to be spammed by so many exception details. If they want to find out what are the concrete exceptions, they can check the debug information.